Sources/Build/BuildPlan.swift

```diff
@@ -499,8 +499,7 @@ public final class SwiftTargetBuildDescription {
 
     /// The path to the swiftmodule file after compilation.
     var moduleOutputPath: AbsolutePath {
-        let dirPath = (target.type == .executable) ? tempsPath : buildParameters.buildPath
-        return dirPath.appending(component: target.c99name + ".swiftmodule")
+        return buildParameters.buildPath.appending(component: target.c99name + ".swiftmodule")
```

Comment:
This restores the code from before a previous change to make it more obvious that tests couldn't import executable modules, by moving those modules into subdirectories of the main build directory.

Reply:
Because Clang targets don't produce a module but instead use a module map in the public headers directory, the more obvious error is preserved when trying to import a Clang executable module from a test target.  It is not preserved when importing an executable module from another one, however.  We should consider moving the module back into its subdirectory and instead adding additional search paths to test targets that declare a dependency on the executable target, to preserve the good error message if an executable tries to import another executable's module.

This would be good to do in general, so that a dependency between two targets needs to be declared before the module can be imported (otherwise it leads to hard-to-diagnose linker errors).